shared/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.js

```diff
@@ -48,242 +53,331 @@ exports.fileAndServeCourtIssuedDocumentInteractor = async (
-          hideFromPendingMessages: true,
-          inProgress: true,
-          section: DOCKET_SECTION,
-          sentBy: user.name,
-          sentByUserId: user.userId,
-        },
-        { applicationContext },
-      );
-    }
+  docketEntryEntity.workItem.assignToUser({
+    assigneeId: user.userId,
+    assigneeName: user.name,
+    section: user.section,
+    sentBy: user.name,
+    sentBySection: user.section,
+    sentByUserId: user.userId,
+  });
 
-    docketEntryEntity.workItem.assignToUser({
-      assigneeId: user.userId,
-      assigneeName: user.name,
-      section: user.section,
-      sentBy: user.name,
-      sentBySection: user.section,
-      sentByUserId: user.userId,
-    });
+  docketEntryEntity.workItem.setAsCompleted({ message: 'completed', user });
 
-    docketEntryEntity.workItem.setAsCompleted({ message: 'completed', user });
+  if (
+    caseEntity.docketEntries.some(
+      docketEntry =>
+        docketEntry.docketEntryId === docketEntryEntity.docketEntryId,
+    )
+  ) {
     caseEntity.updateDocketEntry(docketEntryEntity);
+  } else {
+    caseEntity.addDocketEntry(docketEntryEntity);
+  }
 
-    await applicationContext.getPersistenceGateway().saveWorkItem({
-      applicationContext,
-      workItem: docketEntryEntity.workItem.validate().toRawObject(),
-    });
+  await applicationContext.getPersistenceGateway().saveWorkItem({
+    applicationContext,
+    workItem: docketEntryEntity.workItem.validate().toRawObject(),
+  });
+
+  await applicationContext.getPersistenceGateway().putWorkItemInUsersOutbox({
+    applicationContext,
+    section: user.section,
+    userId: user.userId,
+    workItem: docketEntryEntity.workItem.validate().toRawObject(),
+  });
 
-    await applicationContext.getPersistenceGateway().putWorkItemInUsersOutbox({
+  caseEntity = await applicationContext
+    .getUseCaseHelpers()
+    .updateCaseAutomaticBlock({
       applicationContext,
-      section: user.section,
-      userId: user.userId,
-      workItem: docketEntryEntity.workItem.validate().toRawObject(),
+      caseEntity,
     });
 
-    const { Body: pdfData } = await applicationContext
-      .getStorageClient()
-      .getObject({
-        Bucket: applicationContext.environment.documentsBucketName,
-        Key: docketEntryId,
-      })
-      .promise();
-
-    let serviceStampType = 'Served';
-
-    if (docketEntryEntity.documentType === GENERIC_ORDER_DOCUMENT_TYPE) {
-      serviceStampType = docketEntryEntity.serviceStamp;
-    } else if (
-      ENTERED_AND_SERVED_EVENT_CODES.includes(docketEntryEntity.eventCode)
-    ) {
-      serviceStampType = 'Entered and Served';
-    }
-
-    const serviceStampDate = formatDateString(
-      docketEntryEntity.servedAt,
-      'MMDDYY',
-    );
-
-    const newPdfData = await addServedStampToDocument({
+  if (ENTERED_AND_SERVED_EVENT_CODES.includes(docketEntryEntity.eventCode)) {
+    await closeCaseAndUpdatedTrialSessionForEnteredAndServedDocuments({
```

Comment:
I'm not sure what this is doing.  So are there certain event codes that close the case basically?